algo/src/main/kotlin/com/hartwig/actin/algo/evaluation/molecular/TargetCoveragePredicate.kt

```diff
@@ -3,47 +3,54 @@ package com.hartwig.actin.algo.evaluation.molecular
     2 -> "${plural(this[0])} $conjunction ${plural(this[1])}"
     else -> dropLast(1).joinToString(", ") { plural(it) } + " $conjunction " + plural(last())
 }
 
-fun plural(str: String) = str + "s"
+private fun plural(str: String) = str + "s"
 
-fun any() = or(*MolecularTestTarget.entries.toTypedArray())
+fun any(messagePrefix: String? = null) = or(*MolecularTestTarget.entries.toTypedArray(), messagePrefix = messagePrefix)
 
-fun all() = and(*MolecularTestTarget.entries.toTypedArray())
+fun all(messagePrefix: String) = and(*MolecularTestTarget.entries.toTypedArray(), messagePrefix = messagePrefix)
 
-fun atLeast(target: MolecularTestTarget) = and(target)
+fun atLeast(target: MolecularTestTarget, messagePrefix: String) = and(target, messagePrefix = messagePrefix)
 
-fun and(vararg targets: MolecularTestTarget) = combine(targets.toSet(), Predicate<List<MolecularTestTarget>>::and) {
-    it.map { t -> t.name.lowercase() }.joinWithConjunction("and")
-}
+fun and(vararg targets: MolecularTestTarget, messagePrefix: String) =
+    combine(targets.toSet(), messagePrefix = messagePrefix, Predicate<List<MolecularTestTarget>>::and) {
+        it.map { t -> t.name.lowercase() }.joinWithConjunction("and")
+    }
 
-fun or(vararg targets: MolecularTestTarget): TargetCoveragePredicate = combine(targets.toSet(), Predicate<List<MolecularTestTarget>>::or) {
-    it.map { t -> t.name.lowercase() }.joinWithConjunction("or")
-}
+fun or(vararg targets: MolecularTestTarget, messagePrefix: String? = null): TargetCoveragePredicate =
+    combine(targets.toSet(), messagePrefix = messagePrefix, Predicate<List<MolecularTestTarget>>::or) {
+        it.map { t -> t.name.lowercase() }.joinWithConjunction("or")
+    }
 
 private fun combine(
     targets: Set<MolecularTestTarget>,
+    messagePrefix: String? = null,
     reducer: (Predicate<List<MolecularTestTarget>>, Predicate<List<MolecularTestTarget>>) -> Predicate<List<MolecularTestTarget>>,
     stringify: (Set<MolecularTestTarget>) -> String,
 ): TargetCoveragePredicate = TargetCoveragePredicate(
-    targets, targets.map { target -> Predicate<List<MolecularTestTarget>> { it.contains(target) } }.reduce(reducer), stringify
+    targets,
+    targets.map { target -> Predicate<List<MolecularTestTarget>> { it.contains(target) } }.reduce(reducer),
+    stringify,
+    messagePrefix
 )
 
 class TargetCoveragePredicate(
     private val targets: Set<MolecularTestTarget>,
     private val predicate: Predicate<List<MolecularTestTarget>>,
     private val stringify: (Set<MolecularTestTarget>) -> String,
+    private val messagePrefix: String? = null
 ) :
     Predicate<List<MolecularTestTarget>> {
 
     override fun test(t: List<MolecularTestTarget>) = predicate.test(t)
 
-    override fun toString(): String {
-        return stringify.invoke(targets)
+    fun message(gene: String): String {
+        return "${if (messagePrefix != null) "$messagePrefix " else ""}$gene undetermined (not tested for ${stringify.invoke(targets)})"
```

Comment:
```suggestion
        return "${messagePrefix?.let { "$it "} ?: ""}$gene undetermined (not tested for ${stringify.invoke(targets)})"
```
It would be even better if the prefix included the space 🙂

Reply:
yeah but then everyone who adds a prefix needs to think of that, I'd rather a tiny bit of complexity here to make the caller life easier.

Reply:
The people making changes have to learn the convention either way, and when I first saw you pass in a prefix I thought "Don't we need a space there?" 😛 Of course, it's fine either way.